var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -339,10 +340,14 @@ def gettext_libdir(self):
         # Packages that link to glib were also picking up -lintl from glib's
         # glib-2.0.pc file. However, packages such as py-pygobject were
         # bypassing spack's compiler wrapper for linking and thus not finding
-        # the gettext library directory. The patch below explitly adds the
+        # the gettext library directory. The patch below explicitly adds the
         # appropriate -L path.
         spec = self.spec
-        if spec.satisfies("@2.0:2"):
+        if (
+            spec.satisfies("@2.0:2")
+            and "intl" in self.spec["gettext"].libs.names
+            and not is_system_path(spec["gettext"].libs.directories[0])
```

Comment:
Slight tweak to use `is_system_path(spec["gettext"].prefix)` in the conditional instead of `is_system_path(spec["gettext"].libs.directories[0])` in case `spec["gettext"].libs.directories` is empty.